### PR TITLE
Update docker infra and fix image building and publishing in CI

### PR DESCRIPTION
**Pull request by mjgiarlo**
## Why was this change made?

To update the docker infrastructure for suri-rails and get the image build working again.

## Was the documentation (README, DevOpsDocs, wiki, openapi.yml) updated?

Yes.

## Does this change affect how this application integrates with other services?

Yes, will test in argo, dor-services-app, and hydrus.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -47,25 +55,18 @@ jobs:
           name: Which bundler?
           command: bundle -v
 
-      - restore_cache:
-          keys:
-            - suri-rails-dependencies-{{ checksum "Gemfile.lock" }}
-            # fallback to using the latest cache if no exact match is found
-            - suri-rails-dependencies-
-
```
Comment (ndushay) [inline .circleci/config.yml:55]: we don't want to cache because ...

Comment (mjgiarlo) [reply .circleci/config.yml:55]: @ndushay Good question! This caching is messed up in a few ways. First, it uses different keys for restore and save (dor-services vs. suri-rails). Second, I don't believe the cache is persisted across runs. So, basically, I think this is copypasta from some of our other repos where caching *has* been useful (e.g., in our GoLang and React work), and I'm removing it from Ruby/Rails projects whenever I see it.

**Comment by mjgiarlo** [comment]
marking this as hold until I add some request specs that flag errors like https://github.com/sul-dlss/hydrus/pull/415

File: openapi.yml [hunk 2]
```diff
@@ -84,17 +91,21 @@ paths:
...
+                $ref: '#/components/schemas/ErrorResponse'
       parameters:
         - name: id
           in: path
           description: druid to return
           required: true
           schema:
             $ref: '#/components/schemas/BareDruid'
-  /suri2/namespaces/druids/identifiers:
+  /suri2/namespaces/druid/identifiers:
```
Comment (mjgiarlo) [inline openapi.yml:108]: This was what was causing the Hydrus build to fail, @ndushay, if you were interested. I added a bunch of request specs to this branch to catch these more easily. (This is part of why we've been moving away from controller specs towards request specs!)

**Pull request merged**